modules/Core/pages/api/v2/index.php

```diff
@@ -561,26 +573,23 @@ private function getUserInfo(){
 
             unset($user->id);
 
-            $this->returnArray((array)$user);
+            $this->returnArray((array) $user);
 
         } else $this->throwError(1, $this->_language->get('api', 'invalid_api_key'));
     }
 
     // Set a user's group
-    private function setGroup(){
+    private function setGroup() {
         // Ensure the API key is valid
         if($this->_validated === true) {
-            if(!isset($_POST) || empty($_POST) || !isset($_POST['uuid']) || empty($_POST['uuid']) || !isset($_POST['group_id']) || !is_numeric($_POST['group_id'])){
+            if(!isset($_POST) || empty($_POST) || !isset($_POST['id']) || empty($_POST['id']) || !isset($_POST['group_id']) || !is_numeric($_POST['group_id'])) {
```

Comment:
Are we happy with this being a 2 step process now, first to get the user ID from userInfo then second to set the group?

Reply:
@Derkades

Reply:
The plugin already does 2 requests due to the way the object oriented API is designed. If it's not that much hassle you can add uuid+username support to every API method